client/src/components/pages/RecyclePage/RecyclePage.js

```diff
@@ -23,56 +25,45 @@ class RecyclePage extends Component {
                   mapboxApiAccessToken={TOKEN}
                   mapStyle="mapbox://styles/jreid655/ck452mq2x1o2k1dnw6b76y20v"
                 >
-                {this.props.facilities.map((facility, index)=>(
-                  (this.props.selectedFacility===facility.location_id)?
-                  (<Popup
-                      key={index}
-                      tipSize={5}
-                      anchor="top"
-                      longitude={facility.longitude}
-                      latitude={facility.latitude}
-                      closeOnClick={false}
-                      onClose={this.props.deselectFacility}
-                    >
-                    <div className='RecyclePage-Popup'>
-                      <h1>{facility.description}</h1>
-                      <p>//// TODO Replace dummy info here</p>
-                     <p>Address: ""                </p>
-                     <p>Contact Info: ""           </p>
-                     <p>Materials Accepted: ""     </p>
-                    </div>
-                   </Popup>
-                  ):(<Marker
-                        key={index}
-                        latitude={facility.latitude}
-                        longitude={facility.longitude}
-                        offsetTop={-10}
-                        offsetLeft={-10}>
-                        <img id={facility.location_id}
-                             src={earth}
-                             style={{height:"25px",width:"25px"}}
-                             onClick={() => this.props._onClickMarker(facility)}/>
-                       </Marker>
-                     )))}
+             <MapPin
+               data={this.props.facilities}
+               onClick={this.props.openModalHandler}
+               />
           </ReactMapGL>
         </div>
-          <div className="RecyclePage-Cards">
-            {this.props.facilities.map((facility, index)=>(
-              <Card
-                key={index}
-                img={earth}
-                facility={facility}
-                name={facility.description}
-                // category=""
-                // location=""
-                // description=""
-                // contact=""
-                distance={facility.distance}
-                _onClickCard={this.props._onClickCard}
-                />
+        <div className="RecyclePage-Cards">
+          {this.props.facilities.map((facility, index)=>(
+            <React.Fragment>
+            {(this.props.facilityDetails!==null)?
```

Comment:
This is not required because the modal component itself has true/false to show it or not.

Reply:
I think I added bc I got an error for cases for trying to access data in facility details when it was empty.